lib/hawkular/operations/operations_api.rb

```diff
@@ -128,46 +128,113 @@ def invoke_specific_operation(operation_payload, operation_name, &callback)
       invoke_operation_helper(operation_payload, operation_name, &callback)
     end
 
-    # Deploys a war file into WildFly
+    # Deploys an archive file into WildFly
     #
     # @param [Hash] hash Arguments for deployment
     # @option hash [String]  :resource_path canonical path of the WildFly server into which we deploy
     # @option hash [String]  :destination_file_name resulting file name
     # @option hash [String]  :binary_content binary content representing the war file
-    # @option hash [String]  :enabled whether the deployment should be enabled or not
+    # @option hash [String]  :enabled whether the deployment should be enabled immediately, or not (default = true)
+    # @option hash [String]  :force_deploy whether to replace existing content or not (default = false)
+    # @option hash [String]  :server_groups comma-separated list of server groups for the operation (default = ignored)
```

Comment:
The server_groups is optional and to be supplied when we're dealing with a domain situation.  If not specified then we're talking standalone.  Basically, it is assumed that if server_groups are specified then the WF server performing the command is aware of those server_groups.  You should be able to upload content and enable it (i.e. our default for Add Deployment) whether it's standalone or applied to server groups.  Did I answer the question, I'm not sure?